core/codegen/tests/from_form.rs

```diff
@@ -618,3 +618,33 @@ fn test_nested_multi() {
 //     println!("String: {}", string);
 //     panic!(string);
 // }
+
+#[test]
+fn test_defaults() {
+    #[derive(FromForm, PartialEq, Debug)]
+    struct FormWithDefaults {
+        #[field(default = 100)]
+        field1: i32,
+        field2: i32,
```

Comment:
I suspect you're using `i32` here because inference failed for `usize` or another integer. You should call this out next time. If you try something first and it doesn't work but you expected to, we should figure out how to make it work instead of changing the test or interface to meet our defied expectations.

Reply:
I don't know if the experience is different for you here but the type inference works for all types here so there wasn't anything to call out. The reason I used `i32` is simply that it is the number type that I most usually reach for. I have changed these types to  `usize` and `i128` now, to make sure that this case is tested :+1: